plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ColumnWithAliasFormatter.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.spi.connector.ConnectorSession;
+
+public interface ColumnWithAliasFormatter
```

Comment:
Dropping interface and leaving only a single class. The interface has been introduced only due to the fact that on the previous round of review, that on your request was done via videocall, you specifically requested to break that PR into multiple commits, where one is derived logically from the other. As a result I introduced a step-by-step 4 commit process as if I was guiding you through a refactor session. The natural thing was to introduce an interface first and then swap the implementation later (see commits history to understand better).

It also felt awkward to me that in the end we are left off with only a single implementation of an interface. Maybe the result of that is because there was a miscommunication between two of us WRT how this should be done. I'd appreciate your feedback (if possible more elaborate) on this, so that I can understand better what went wrong here. This will allow me to avoid doing unnecessary work in the future.

With the OptionalInt parameter, I'm withholding this change for now, as I still don't know how to code against that in the integration test, so that change is coupled with the other bulky work I need to do anyway.